In [1]:
#Importing pandas library to read CSV data file
import pandas as pd

In [3]:
#Reading CSV data file in Python
dataset = pd.read_csv('bank-data.csv')

In [4]:
dataset

AGE HOME  INCOME GENDER  HOUSEHOLD_N  CREDIT_LINES_N  DEAFULTED
0      23    R       2      M            3               2          1
1      43    R       3      F            2               2          1
2      63    R       7      M            3               3          0
3      46    H       7      F            4               4          0
4      55    H       7      F            3               4          0
...   ...  ...     ...    ...          ...             ...        ...
1095   24    R       5      F            5               4          1
1096   44    H       5      M            3               3          0
1097   67    H       5      F            1               1          0
1098   35    R       6      F            3               0          0
1099   34    H       5      M            3               1          0

[1100 rows x 7 columns]

In [5]:
#Dividing dataset into X and y
X = dataset.iloc[:, [0,2,4,5]].values
y = dataset.iloc[:, -1].values

In [6]:
X

array([[23,  2,  3,  2],
       [43,  3,  2,  2],
       [63,  7,  3,  3],
       ...,
       [67,  5,  1,  1],
       [35,  6,  3,  0],
       [34,  5,  3,  1]], dtype=int64)

In [7]:
y

array([1, 1, 0, ..., 0, 0, 0], dtype=int64)

In [8]:
#Importing train_test_split from sklearn.model_selection to split data into training and testing sets
from sklearn.model_selection import train_test_split
X_train, X_test, y_train,  y_test = train_test_split(X,y,test_size=0.2,random_state=0)

In [21]:
len(y_train)

880

In [13]:
#Importing StandardScaler from sklearn.preprocessing to scale matrix of features
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)

In [14]:
#Importing LogisticRegression from sklearn.linear_model to build LogisticRegression classifier
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(random_state = 0)
classifier.fit(X_train, y_train)

d:\gitrepositories\machinelearning\ml_env\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=0, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [15]:
y_pred = classifier.predict(X_test)

In [17]:
y_pred

array([0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0,
       0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1,
       1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0,
       0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0,
       1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 0, 1, 0, 1,
       0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0,
       0, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1,
       1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0,
       1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0,
       0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0],
      dtype=int64)

In [18]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)

In [19]:
cm

array([[105,  37],
       [ 17,  61]], dtype=int64)

### The outcome of the Confusion matrix says that our model predicted (105 + 61) = 166 times and false predicted (37 + 17) = 54 times. Now the percentage of accuracy by this model is 166 / 220 (i.e., y_test) * 100 = 75.45%. This fairly good as its above 70%, and still it has room for improvement.